### PR TITLE
feature(gce): block external network for test resources

### PR DESCRIPTION
**Pull request by soyacz**
Apply better firewall rules to close external network access.

closes: https://github.com/scylladb/qa-tasks/issues/1424

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by soyacz** [comment]
test run in multi-dc cluster: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/features-google-cloud-snitch-mutli-dc-lukasz/2/

I couldn't communicate using public IP's, still test works.

Remaining things to cover:
1. builders need to add `sct-network` network tag to be able to communicate with db nodes (required for artifact test)
2. no ssh connection to db nodes: I see 2 resolutions: add `hydra ssh` support for GCE (effort required), or create command to unlock just ssh port (easy one) on demand
3. current firewall rules can be tuned to this approach so we secure machines even better

File: sdcm/sct_runner.py [hunk 1]
```diff
@@ -797,7 +797,7 @@ def _create_instance(self,
                                        "block-project-ssh-keys": "true",
                                        "ssh-keys": f"{self.LOGIN_USER}:{self.key_pair.public_key.decode()}",
                                    },
-                                   network_tags=["keep-alive"],
+                                   network_tags=["keep-alive", "sct-network"],
```
Comment (soyacz) [inline sdcm/sct_runner.py:800]: btw. @fruch is it intended to set keep-alive as network tag?

**Comment by soyacz** [comment]
test run in multi-dc cluster: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/features-google-cloud-snitch-mutli-dc-lukasz/2/

I couldn't communicate using public IP's, still test works.

**Review by fgelcer — APPROVED** [review]
LGTM

**Comment by fgelcer** [comment]
> test run in multi-dc cluster: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/features-google-cloud-snitch-mutli-dc-lukasz/2/
> 
> I couldn't communicate using public IP's, still test works.

the failure you see in here, during cluster health check was caused by a change i did recently, and should be fixed by #6610

File: sdcm/sct_runner.py (hunk 1 above)
Comment (fruch) [reply sdcm/sct_runner.py:800]: Yes, it's still how the GCE resource cleaner code is looking it up.

Once we'll update that code we'll be able to remove this network tag

**Comment by soyacz** [comment]
> > test run in multi-dc cluster: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/features-google-cloud-snitch-mutli-dc-lukasz/2/
> > I couldn't communicate using public IP's, still test works.
> 
> the failure you see in here, during cluster health check was caused by a change i did recently, and should be fixed by #6610

@fgelcer Still I see these failures, are you sure it is fixed? See artifact test I tried: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/artifacts-gce-image-test/9/

**Comment by soyacz** [comment]
> > > > test run in multi-dc cluster: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/features-google-cloud-snitch-mutli-dc-lukasz/2/
> > > > I couldn't communicate using public IP's, still test works.
> > > 
> > > 
> > > the failure you see in here, during cluster health check was caused by a change i did recently, and should be fixed by #6610
> > 
> > 
> > @fgelcer Still I see these failures, are you sure it is fixed? See artifact test I tried: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/artifacts-gce-image-test/9/
> 
> Are you sure you have the fix in ?

yes, I'm sure: https://github.com/soyacz/scylla-cluster-tests/commits/qa-1424-close-external-network-gce

**Comment by fruch** [comment]
> > > > > test run in multi-dc cluster: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/features-google-cloud-snitch-mutli-dc-lukasz/2/
> > > > > I couldn't communicate using public IP's, still test works.
> > > > 
> > > > 
> > > > the failure you see in here, during cluster health check was caused by a change i did recently, and should be fixed by #6610
> > > 
> > > 
> > > @fgelcer Still I see these failures, are you sure it is fixed? See artifact test I tried: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/artifacts-gce-image-test/9/
> > 
> > 
> > Are you sure you have the fix in ?
> 
> yes, I'm sure: https://github.com/soyacz/scylla-cluster-tests/commits/qa-1424-close-external-network-gce

@soyacz, in you case, I might think it's cause of a different reason.
`curl: (6) Could not resolve host`

the address is 127.0.0.1, not sure why it can't be resolved...

**Comment by soyacz** [comment]
I posted another fix for health check validation: https://github.com/scylladb/scylla-cluster-tests/pull/6613

**Comment by fgelcer** [comment]
> I posted another fix for health check validation: #6613

sorry, my bad :-(

**Comment by soyacz** [comment]
verified here: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/artifacts-gce-image-test/11/
seems working

**Comment by fruch** [comment]
> verified here: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/artifacts-gce-image-test/11/ seems working

we verified it with a longevity / rolling upgrade ?

**Comment by soyacz** [comment]
> > verified here: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/artifacts-gce-image-test/11/ seems working
> 
> we verified it with a longevity / rolling upgrade ?

longevity: https://jenkins.scylladb.com/job/scylla-staging/job/lukasz/job/longevity-mini-test-1h-test/31/

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
@soyacz this seems also save to backport,
we'll let it sock for a week, and then we'll backport

**Comment by vponomaryov** [comment]
2/12 rolling upgrade CI jobs from failed with session timeouts which were run in parallel.
In both logs there are sighs of remote unknown connections.

So, May I backport it to `branch-5.2`. Because, anyway I am the person who will investigate it's possible failures.
@fruch ? @roy ?

**Comment by fruch** [comment]
@soyacz let's backport to both 5.2 and  2023.1

**Comment by soyacz** [comment]
seems it requires work from https://github.com/scylladb/scylla-cluster-tests/pull/6137
I can try doing it using libcloud (there's `ex_tags` which I believe is equivalent of network-tags, but I'm not sure), or fight with 6137 backport

**Comment by fruch** [comment]
> seems it requires work from #6137 I can try doing it using libcloud (there's `ex_tags` which I believe is equivalent of network-tags, but I'm not sure), or fight with 6137 backport

if so, leave it for now.

the libcloud removal is too big to backport.

**Pull request merged**